be/src/storage/vectorized/schema_change.cpp

```diff
@@ -0,0 +1,1679 @@
+        }
+
+        std::unique_ptr<RowsetWriter> rowset_writer;
+        OLAPStatus st = RowsetFactory::create_rowset_writer(writer_context, &rowset_writer);
+        if (st != OLAP_SUCCESS) {
+            status = Status::InternalError("build rowset writer failed");
+            LOG(INFO) << "build rowset writer failed";
+            goto PROCESS_ALTER_EXIT;
+        }
+
+        if (!sc_procedure->process(sc_params.rowset_readers[i], rowset_writer.get(), new_tablet, base_tablet,
+                                   sc_params.rowsets_to_change[i])) {
+            LOG(WARNING) << "failed to process the version."
+                         << " version=" << sc_params.version.first << "-" << sc_params.version.second;
+            status = Status::InternalError("process failed");
+            goto PROCESS_ALTER_EXIT;
+        }
+        // Add the new version of the data to the header,
+        // To prevent deadlocks, be sure to lock the old table first and then the new one
+        sc_params.new_tablet->obtain_push_lock();
+        RowsetSharedPtr new_rowset = rowset_writer->build();
+        if (new_rowset == nullptr) {
+            LOG(WARNING) << "failed to build rowset, exit alter process";
+            sc_params.new_tablet->release_push_lock();
+            goto PROCESS_ALTER_EXIT;
+        }
+        LOG(INFO) << "new rowset has " << new_rowset->num_segments() << " segments";
+        OLAPStatus res = sc_params.new_tablet->add_rowset(new_rowset, false);
+        if (res == OLAP_ERR_PUSH_VERSION_ALREADY_EXIST) {
+            LOG(WARNING) << "version already exist, version revert occured. "
+                         << "tablet=" << sc_params.new_tablet->full_name() << ", version='" << sc_params.version.first
+                         << "-" << sc_params.version.second;
+            StorageEngine::instance()->add_unused_rowset(new_rowset);
+            status = Status::OK();
+        } else if (res != OLAP_SUCCESS) {
+            LOG(WARNING) << "failed to register new version. "
+                         << " tablet=" << sc_params.new_tablet->full_name() << ", version=" << sc_params.version.first
+                         << "-" << sc_params.version.second;
+            StorageEngine::instance()->add_unused_rowset(new_rowset);
+            sc_params.new_tablet->release_push_lock();
+            goto PROCESS_ALTER_EXIT;
+        } else {
+            VLOG(3) << "register new version. tablet=" << sc_params.new_tablet->full_name()
+                    << ", version=" << sc_params.version.first << "-" << sc_params.version.second;
+        }
+        sc_params.new_tablet->release_push_lock();
+
+        VLOG(10) << "succeed to convert a history version."
+                 << " version=" << sc_params.version.first << "-" << sc_params.version.second;
+        // XXX: The SchemaChange state should not be cancelled at this point,
+        // because the new Delta has to be converted to the old and new Schema versions
+    }
+
+PROCESS_ALTER_EXIT : {
```

Comment:
you can DeferOp to replace PROCESS_ALTER_EXIT and goto statement

Reply:
done